interfaces/builtin/accounts_control.go

```diff
@@ -0,0 +1,97 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const accountsControlConnectedPlugAppArmor = `
+# Allow setting the password for users and groups and modifying
+# passwd and shadow entries
+/{,usr/}bin/passwd ixr,
+/{,usr/}bin/gpasswd ixr,
+/{,usr/}bin/chage ixr,
+/{,usr/}bin/chfn ixr,
+/{,usr/}bin/chsh ixr,
+/{,usr/}bin/expiry ixr,
+/{,usr/}sbin/unix_chkpwd ixr,
+
+# Also allow the batch-mode variants
+/{,usr/}sbin/chpasswd ixr,
+/{,usr/}sbin/chgpasswd ixr,
+/{,usr/}sbin/newusers ixr,
+
+# Allow checking integrity of files
+/{,usr/}sbin/pwck ixr,
+/{,usr/}sbin/grpck ixr,
+
+# Allow adding, removing and modifying users and groups
+/{,usr/}sbin/{add,del}user ixr,
+/{,usr/}sbin/user{add,mod,del} ixr,
+/{,usr/}sbin/{add,del}group ixr,
+/{,usr/}sbin/group{add,mod,del} ixr,
+
+# Only allow modifying the non-system extrausers database
+/var/lib/extrausers/{,**} rwk,
+
+# needed by {add,del}user
+/etc/login.defs r,
+/etc/{add,del}user.conf r,
+/proc/*/loginuid r,
+/etc/default/{useradd,nss} r,
+/etc/pam.d/{{,ch}passwd,common-{account,password,auth,session},other} r,
+
+/etc/.pwd.lock w,
+/etc/gshadow r,
+/etc/{passwd,group,gshadow}* wl,
+/etc/{sub{u,g}id,shadow} rw,
+/etc/{sub{u,g}id,shadow}.lock l,
+/etc/{sub{u,g}id,shadow}.* w,
```

Comment:
All of these write accesses I would expect you to need 'k', but most importantly, they are granting access to update the system files in /etc instead of only the files in /var/lib/extrausers/. This interface is meant to only allow updating /var/lib/extrausers/ (see "Only allow modifying the non-system extrausers database" comment from above). What tools prompted these rules? They should be updated to use extrausers and if they cannot, please file bugs and talk to ogra.